### ISSUE TITLE
Create a test suite

### ISSUE THREAD

**Issue by Philonous** [pontarius/pontarius-service#5]
We need a way to run tests.

It should be one command (possibly a script).

**Comment by sils** [comment]
Hey, this is Lasse from GUADEC :) You probably want continuous integration, you can take a look at what we at coala do: https://github.com/coala-analyzer/coala/ (I don't think I'm arrogant if I say we are testing above average with 100% branch coverage. A good symptom of you testing your stuff well is that, if master breaks usually it's not your fault but the CI hosters or so ;))

I can maybe help you a bit setting up though I'll be not really around during next two weeks.

**Comment by jonkri** [comment]
Hi, Lasse!

I hope that everything is well with you!

Thanks for the tip about CI! :) We've been thinking about that, actually. I must admit, though, I've never seen a project as well covered as yours, with all those CI systems. :)

**Comment by sils** [comment]
before you set this up: switch to gitlab! makes everything easier and CI locks you in :)

**Comment by jonkri** [comment]
We've actually been thinking about GitLab. Is it possible to live without GitHub, though? :)

I'm not sure what you mean by locking in... Do you mean that GitHub locks you in?

**Comment by sils** [comment]
Yes it's possible. You can set up GitLab so it auto mirrors to GitHub and use GitMate (gitmate.io, a service I'm currently writing based on coala) to set up an autoresponse on GitHub. https://gitlab.com/gitlab-org/gitlab-ce/issues/17271 is blocking the mirroring part for now but they should solve that soon.

GitHub locks you in, when you start using GitHub you wil come to rely on certain things: your stars are a valuable thing to promote your projects, you switch, you loose them. The CI needs to be redone on another platform as well. Those are certain factors preventing you from switching to another platform and thus locking you in. If you want to switch rather do it earler than later.

**Comment by jonkri** [comment]
Thanks for the advice! However, looking at the GitMate web site, it's not entirely clear exactly what it does. What kind of automatic responses are you referring to here? I think it would be beneficial to clearly list the top specific problems that your service is solving.

**Comment by sils** [comment]
Yeah, we're working on that. GitMate is primarily a code review bot (using coala) but it has an autoresponse feature as well which is why I recommended it for you.
